R/process_lookup_gpprac.R

```diff
@@ -0,0 +1,123 @@
+#' Process the SLF gpprac lookup
+#'
+#' @description This will read and process the
+#' gpprac lookup, it will return the final data
+#' but also write this out as a zsav and rds.
+#'
+#' @param write_to_disk (optional) Should the data be written to disk default is
+#' `TRUE` i.e. write the data to disk.
+#'
+#' @return the final data as a [tibble][tibble::tibble-package].
+#' @export
+#' @family process extracts
+
+process_lookup_gpprac <- function(write_to_disk = TRUE) {
+
+  # Read lookup files -------------------------------------------------------
+
+  # Retrieve the latest resource from the dataset
+  opendata <-
+    phsopendata::get_dataset("gp-practice-contact-details-and-list-sizes",
+      max_resources = 20
+    ) %>%
+    janitor::clean_names() %>%
+    dplyr::left_join(
+      phsopendata::get_resource(
+        "944765d7-d0d9-46a0-b377-abb3de51d08e",
+        col_select = c("HSCP", "HSCPName", "HB", "HBName")
+      ) %>%
+        janitor::clean_names(),
+      by = c("hb", "hscp")
+    ) %>%
+    # select variables
+    dplyr::select(
+      gpprac = .data$practice_code,
+      practice_name = .data$gp_practice_name,
+      .data$postcode,
+      cluster = .data$gp_cluster,
+      partnership = .data$hscp_name,
+      health_board = .data$hb_name
+    ) %>%
+    # drop NA cluster rows
+    tidyr::drop_na(.data$cluster) %>%
+    # format practice name text
+    dplyr::mutate(practice_name = stringr::str_to_title(.data$practice_name)) %>%
+    # format postcode
+    dplyr::mutate(postcode = phsmethods::format_postcode(.data$postcode)) %>%
+    # keep distinct gpprac
+    dplyr::distinct(.data$gpprac, .keep_all = TRUE) %>%
+    # Sort for SPSS matching
+    dplyr::arrange(.data$gpprac) %>%
+    # Write rds file
+    write_rds(get_practice_details_path(check_mode = "write"))
+
+
+  # Read Lookup files ---------------------------------------
+  # gp lookup
+  gpprac_ref_file <-
+    readr::read_rds(get_gpprac_ref_path()) %>%
```

Comment:
The test run reports an error there. It is because the function `get_gpprac_ref_path()` use the extension '.sav'. I guess it will be changed to '.rds' some while later but for now, we use '.sav'. Am I right?

Reply:
Good spot, i'll push the change and we can read the zsav version for now but this will change to .rds when this is available

Reply:
Can we check if there's tests for `get_gpprac_ref_path` so we can be alerted if/when it's changed from a sav to rds version.

Reply:
This is now done in #446